examples/src/main/scala/example/akkahttp/AuthExampleApp.scala

```diff
@@ -2,37 +2,37 @@ package example.akkahttp
 
 import akka.actor.ActorSystem
 import akka.http.scaladsl.Http
-import akka.http.scaladsl.model.{ HttpResponse, StatusCodes }
 import akka.http.scaladsl.server.Directives.{ getFromResource, path, _ }
-import akka.http.scaladsl.server.RequestContext
-import caliban.AkkaHttpAdapter.ContextWrapper
 import caliban.GraphQL._
-import caliban.RootResolver
-import caliban.interop.circe.AkkaHttpCirceAdapter
+import caliban.interop.tapir.RequestInterceptor
 import caliban.schema.GenericSchema
+import caliban.{ AkkaHttpAdapter, RootResolver }
+import sttp.model.StatusCode
+import sttp.tapir.json.circe._
+import sttp.tapir.model.ServerRequest
 import zio.blocking.Blocking
 import zio.internal.Platform
 import zio.random.Random
-import zio.{ FiberRef, Has, RIO, Runtime, URIO, ZIO }
+import zio.{ FiberRef, Has, RIO, Runtime, ZIO }
 
 import scala.concurrent.ExecutionContextExecutor
 import scala.io.StdIn
 
-object AuthExampleApp extends App with AkkaHttpCirceAdapter {
+object AuthExampleApp extends App {
 
   case class AuthToken(value: String)
 
   type Auth = Has[FiberRef[Option[AuthToken]]]
 
-  object AuthWrapper extends ContextWrapper[Auth, HttpResponse] {
-    override def apply[R <: Auth, A >: HttpResponse](
-      ctx: RequestContext
-    )(effect: URIO[R, A]): URIO[R, A] =
-      ctx.request.headers.collectFirst {
+  object AuthInterceptor extends RequestInterceptor[Auth] {
+    override def apply[R <: Auth](
+      request: ServerRequest
+    ): ZIO[R, StatusCode, Unit] =
+      request.headers.collectFirst {
         case header if header.is("token") => header.value
       } match {
-        case Some(token) => ZIO.accessM[Auth](_.get.set(Some(AuthToken(token)))) *> effect
-        case _           => ZIO.succeed(HttpResponse(StatusCodes.Forbidden))
+        case Some(token) => ZIO.accessM[Auth](_.get.set(Some(AuthToken(token))))
+        case _           => ZIO.fail(StatusCode.Forbidden)
       }
```

Comment:
I think this is a good semantic unifications, sending the errors down the explicit error channel.